src/canvas.rs

```diff
@@ -695,6 +695,13 @@ impl Painter {
             }
         })?;
 
+        if let Some(updated_current_widget) = app_state
+            .widget_map
+            .get(&app_state.current_widget.widget_id)
+        {
+            app_state.current_widget = updated_current_widget.clone();
+        }
+
```

Comment:
TIL that the old way we did bounds didn't actually update the "current" widget, since it's a *clone* of the widget.  

I have no idea how the old version worked without me doing this.